registry/registry.go

```diff
@@ -29,6 +30,9 @@ import (
 // used in a Deployment Manager configuration. There can be multiple
 // registry implementations.
 type Registry interface {
+	// Also handles http.Client.Do method for authenticated File accesses
+	util.HTTPDoer
```

Comment:
Should Registry expose raw http verbs via Do, or should it expose strongly typed methods that get, create, update and delete files?